admin/test/test_packaging.py

```diff
@@ -1246,6 +1289,32 @@ def test_build_command(self):
         """
         self.assertIs(build_in_docker, BuildScript.build_command)
 
+    def test_default_distributions(self):
+        """
+        ``BuildScript.main`` uses the ``admin/build_targets/`` directory next
+        to the ``admin.packaging`` module it comes from to determine supported
+        distributions if no ``top_level`` path is passed to it.
+        """
+        supported_distribution = FLOCKER_PATH.descendant([
+            b"admin", b"build_targets"
+        ]).children()[0].basename()
+
+        fake_sys_module = FakeSysModule(argv=[
+            b"build-command-name",
+            b"--distribution", supported_distribution,
+            b"http://www.example.com/foo/bar.whl",
+        ])
+
+        def record_arguments(distribution, *args, **kwargs):
+            self.distribution = distribution
+            return SpyStep()
+
+        script = BuildScript(sys_module=fake_sys_module)
+        script.build_command = record_arguments
+        script.main()
+
+        self.assertEqual(self.distribution, supported_distribution)
+
```

Comment:
This test implies that there is non-testing case where ``top_level`` is not used?